test/modules/fetch/intercept/fetch.test.ts

```diff
@@ -329,3 +330,22 @@ it('intercepts an HTTPS PATCH request', async () => {
 
   expect(requestId).toMatch(REQUEST_ID_REGEXP)
 })
+
+it.only('support content-encoding: gzip, br', async () => {
+  const message = 'Lorem ipsum dolor sit amet'
+  const compressed = zlib.brotliCompressSync(zlib.gzipSync(message))
+
+  interceptor.once('request', ({ controller }) => {
+    controller.respondWith(new Response(compressed, {
```

Comment:
For mocked responses, you most definitely need a third-party. Interceptors mustn't meddle with your input based on any headers. 

I'm mostly curious about _receiving_ a compressed request in Node.js.

Reply:
> I'm mostly curious about receiving a compressed request in Node.js.

I hope I understand your question correctly but:
Fetch: decompress the response according to the content encoding header.
IncomingMessage: return the body as-is

Reply:
Got it! That seems to be the case, although we may want to add integration tests for the `ClientRequest` interceptor to verify it. I do believe it's going to expose the response body as-is and it's up to the developer to decompress it. 

The fetch client has decompression built-in, I can confirm it by looking at the [source](https://github.com/nodejs/undici/blob/c1f7d2a9a6556e5a34fdf5ed38c77ce636f1ff02/lib/web/fetch/index.js#L2139-L2161) of Undici.